src/ray/ray_syncer/ray_syncer_bidi_reactor_base.h

```diff
@@ -47,16 +53,24 @@ class RaySyncerBidiReactorBase : public RaySyncerBidiReactor, public T {
       instrumented_io_context &io_context,
       std::string remote_node_id,
       std::function<void(std::shared_ptr<const RaySyncMessage>)> message_processor,
+      std::function<void(RaySyncerBidiReactor *, bool)> cleanup_cb,
       size_t max_batch_size,
       uint64_t max_batch_delay_ms)
       : RaySyncerBidiReactor(std::move(remote_node_id)),
         io_context_(io_context),
-        message_processor_(std::move(message_processor)),
+        message_processor_(message_processor),
+        cleanup_cb_(cleanup_cb),
```

Comment:
Why change the move semantics here? isn't this going to create another copy?

Reply:
`std::function` is just a pointer anyways -- there's no difference w/ move, so it's best practice to just copy

Reply:
interesting, it is! I always assumed the function obj would have some heap-allocated struct containing the arguments. but I guess an std::function can be stack allocated, so heap-allocating the arguments in that case would be strange